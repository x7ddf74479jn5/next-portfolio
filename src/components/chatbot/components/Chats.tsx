import List from "@material-ui/core/List";
import { createStyles, makeStyles } from "@material-ui/core/styles";
import React from "react";

import type { Chat as ChatType } from "../../../types/chatbot/index";
import { Chat } from "./index";

const useStyles = makeStyles(() => {
  return createStyles({
    chats: {
      height: 400,
      padding: "0",
      overflow: "auto",
    },
  });
});

type Props = { chats: ChatType[] };

const Chats: React.VFC<Props> = (props) => {
  const classes = useStyles();

  return (
    <List className={classes.chats} id={"scroll-area"}>
      {props.chats.map((chat, index) => {
        return <Chat text={chat.text} type={chat.type} key={index.toString()} />;
      })}
    </List>
  );
};

export default Chats;
