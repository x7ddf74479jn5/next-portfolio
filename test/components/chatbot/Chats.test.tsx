import Chats from "src/components/chatbot/Chats";
import type { Chat } from "src/types/chatbot";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("Chats", () => {
  let renderResult: RenderResult;
  const data: Chat[] = [
    {
      text: "text",
      type: "question",
    },
    {
      text: "text",
      type: "answer",
    },
  ];

  describe("question", () => {
    beforeEach(() => {
      renderResult = render(<Chats chats={data} />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      expect(renderResult.getByRole("list")).toBeInTheDocument();
      expect(renderResult.getByRole("list")).toHaveAttribute("class", expect.stringContaining("chats"));
      expect(renderResult.getByRole("list")).toHaveAttribute("id", expect.stringContaining("scroll-area"));
      expect(renderResult.getByRole("list").childElementCount).toBe(2);
    });
  });
});
