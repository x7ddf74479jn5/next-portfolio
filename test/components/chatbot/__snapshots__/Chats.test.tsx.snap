// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chats question matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root makeStyles-chats-1 MuiList-padding"
    id="scroll-area"
  >
    <li
      class="MuiListItem-root chatRow MuiListItem-gutters"
    >
      <div
        class="MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle"
        >
          <img
            alt="icon"
            class="MuiAvatar-img"
            src="/img/icons/pandashark_180-180.png"
          />
        </div>
      </div>
      <div
        class="chatBubble"
      >
        text
      </div>
    </li>
    <li
      class="MuiListItem-root chatReverse MuiListItem-gutters"
    >
      <div
        class="MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle"
        >
          <img
            alt="icon"
            class="MuiAvatar-img"
            src="/img/icons/no-profile.png"
          />
        </div>
      </div>
      <div
        class="chatBubble"
      >
        text
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
